pkg/trace/api/normalizer.go

```diff
@@ -32,37 +33,76 @@ var (
 )
 
 // normalize makes sure a Span is properly initialized and encloses the minimum required info
-func normalize(s *pb.Span) error {
+func normalize(ts *info.TagStats, s *pb.Span, firstSpan *pb.Span) error {
+	if s.TraceID == 0 {
+		atomic.AddInt64(&ts.TracesDroppedTraceIdZero, 1)
+		log.Debugf("dropping invalid trace (reason:trace_id_zero): %s", s)
+		return errors.New("trace_id_zero")
+	}
+	if s.SpanID == 0 {
+		atomic.AddInt64(&ts.TracesDroppedSpanIdZero, 1)
+		log.Debugf("dropping invalid trace (reason:span_id_zero): %s", s)
+		return errors.New("span_id_zero")
+	}
+	if s.TraceID != firstSpan.TraceID {
```

Comment:
The signature of this function becomes confusing now. Why can't we leave this comparison where it was? In the `for` loop? It seems redundant to pass what is basically a trace ID to a function, as an entire span, only to compare the current span's trace ID to it.

Reply:
Good point, on second thought this does seem confusing. Moved the foreign span check to outside `normalize` so we don't need to pass the firstSpan in anymore.